src/components/RadioButton/RadioButton.st.css

```diff
@@ -198,10 +198,26 @@
 
 /* Create the indicator (the dot/circle - hidden when not checked) */
 
+.root:box:hover:checked .wrapper {
+    border-color: applyOpacity(value(CurrentIconColor), 1);
+}
```

Comment:
I'm not sure this is needed, I think that it's always defined without opacity when it's `checked`